messaging/WFLY-13793_disable_legacy_amq1_prefix.adoc

```diff
@@ -0,0 +1,91 @@
+= [WFLY-13793] Allow for a remote jms queue / topic not to use legacy amq1 prefix
+:author:            Chao Wang
+:email:             chaowan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:          messaging,jms
+:idseparator:       -
+
+== Overview
+Currently we can disable the use of legacy amq1 prefixes `jms.queue` and `jms.topic` only at the cnnection factory level but not for queue or topic defined in the model or using JakartaEE annotations.
```

Comment:
@soul2zimate This dives directly into the details; I am having difficulties to figure out the whole picture;

Can you add details about the following? 

- which are all the points in the WildFly configurations where it is possible to configure the legacy prefixes? 
- how the setting on the connection factory and on queue and topic work together? is there some kind of overlapping?
- how does a complete configuration using legacy prefixes looks like? (also the Artemis part would be useful)
- how does a complete configuration disabling using legacy prefixes looks like? (also the Artemis part would be useful)

The goal is to have a complete picture of what is the current situation and how it's going to change and, of course, being able to do the setup on my local PC

Reply:
@soul2zimate typo "cnnection"

Reply:
@soul2zimate can you add a quick explain why the attribute at the pooled connection factory level doesn't work? I mean, what is its purpose then and why it doesn't work since everything goes through the pooled connection factory ?

Reply:
It doesn't work because the managing API of Artemis doesn't take the prefix into account when creating a destination. The pcf prefix only have an affect on the destination which is being listened. And the new external jms queue/topic prefix is configured for the destination is being created.